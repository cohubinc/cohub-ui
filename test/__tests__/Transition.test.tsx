import React from "react";
import { mount } from "enzyme";

import { Transition } from "dist";

describe("Transition", () => {
  it("renders show without crashing", async () => {
    const wrapper = mount(
      <Transition show>
        <span>show me</span>
      </Transition>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without show without crashing", async () => {
    const wrapper = mount(
      <Transition show={false}>
        <span>don't show me</span>
      </Transition>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
