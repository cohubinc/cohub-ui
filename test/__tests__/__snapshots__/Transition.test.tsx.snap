// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transition renders show without crashing 1`] = `
<Transition
  show={true}
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={true}
  >
    <span
      key="0/.0"
      style={
        Object {
          "transition": "all 300ms ease-in-out",
          "transitionProperty": "all",
        }
      }
    >
      show me
    </span>
  </Transition>
</Transition>
`;

exports[`Transition renders without show without crashing 1`] = `
<Transition
  show={false}
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={true}
  />
</Transition>
`;
